Automattic/wp-calypso
# State: Remove notices middleware

#### Changes proposed in this Pull Request

In previous PRs we've mostly removed the notices middleware. As of today, it only has a single usage, and this PR migrates it. It also removes the notices middleware altogether.

We still have to migrate and remove `withoutNotice`, but I'll be doing that in another PR.

#### Testing instructions

* Sandbox the API and enable the `DISABLE_WPCOM_GUIDED_TRANSFER` constant (it's currently disabled).
* Go to `/export/guided/:site`
* Pick one of the hosts.
* Input valid details into the form.
* Verify you still get a success notice.
* Verify tests still pass.
